DayPicker/DaySelectorDecorator.js

```diff
@@ -203,113 +208,100 @@ function getSelectedDayString (selected, noneText = '', dayNameLength = 'long')
-			disabled: PropTypes.bool,
-
-			/**
-			 * Current locale.
-			 *
-			 * @type {String}
-			 * @public
-			 */
-			locale: PropTypes.string,
-
-			/**
-			 * Called when an day is selected or unselected.
-			 *
-			 * The event payload will be an object with the following members:
-			 * * `selected` - An array of numbers representing the selected days, 0 indexed
-			 * * `content` - Localized string representing the selected days
-			 *
-			 * @type {Function}
-			 * @public
-			 */
-			onSelect: PropTypes.func,
-
-			/**
-			 * An array of numbers (0-indexed) representing the selected days of the week.
-			 *
-			 * @type {Number|Number[]}
-			 * @public
-			 */
-			selected: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)])
-		};
-
-		static defaultProps = {
-			dayNameLength: 'long',
-			disabled: false
-		};
+	const DaySelector = (props) => {
+		const daySelectorDecoratorProps = setDefaultProps(props, daySelectorDecoratorDefaultProps);
+		const {dayNameLength, locale, selected, ...rest} = daySelectorDecoratorProps;
 
-		handleSelect = ({selected}) => {
-			const {dayNameLength, locale} = this.props;
-			const state = getLocaleState(dayNameLength, locale);
+		const state = getLocaleState(dayNameLength, locale);
+		const localSelected = localizeSelected(selected, state);
+		const abbreviatedDayNames = orderDays(state.abbreviatedDayNames, state);
+		const fullDayNames = orderDays(state.fullDayNames, state);
 
+		const handleSelect = useCallback(({selected: selectedDay}) => {
 			// adjust the selected value beforehand so getSelectedDayString always operates on the
 			// standard, "Sunday as index 0" format
-			selected = generalizeSelected(selected, state);
-			const content = getSelectedDayString(selected, '', dayNameLength);
-
-			forwardCustom('onSelect', () => ({selected, content}))(null, this.props);
-		};
+			const generalSelected = generalizeSelected(selectedDay, state);
+			const content = getSelectedDayString(generalSelected, '', dayNameLength);
+
+			forwardCustom('onSelect', () => ({selected: generalSelected, content}))(null, daySelectorDecoratorProps);
+		}, [dayNameLength, daySelectorDecoratorProps, state]);
+
+		return (
+			<Wrapped
+				{...rest}
+				onSelect={handleSelect}
+				selected={localSelected}
+			>
+				{abbreviatedDayNames.map((children, index) => ({
+					children,
+					// "short" dayNameLength can result in the same name so adding index
+					key: `${index} ${children}`,
+					'aria-label': fullDayNames[index]
+				}))}
+			</Wrapped>
+		);
+	};
 
-		render () {
-			const {dayNameLength, locale, selected, ...rest} = this.props;
-			const state = getLocaleState(dayNameLength, locale);
-
-			const localSelected = localizeSelected(selected, state);
-			const abbreviatedDayNames = orderDays(state.abbreviatedDayNames, state);
-			const fullDayNames = orderDays(state.fullDayNames, state);
-
-			delete rest.everyDayText;
-			delete rest.everyWeekdayText;
-			delete rest.everyWeekendText;
```

Comment:
please leave a comment in the PR description about why these are not needed anymore. i see that they are not present as props